src/LinuxTracing/UprobesUnwindingVisitorDwarfUnwindingTest.cpp

```diff
@@ -1805,10 +1808,18 @@ REGISTER_TYPED_TEST_SUITE_P(
     VisitTwoValidStackSamplesSendsAddressInfosOnlyOnce,
     VisitValidStackSampleWithNullptrMapInfosSendsCompleteCallstackAndAddressInfosWithoutModuleName);
 
+template <typename T, typename U>
+struct DwarfUnwindingTestType {
+  using PerfEventT = T;
+  using ProducerCaptureEventT = U;
+};
+
 using TestTypes = ::testing::Types<
-    std::pair<StackSamplePerfEvent, orbit_grpc_protos::FullCallstackSample>,
-    std::pair<SchedWakeupWithStackPerfEvent, orbit_grpc_protos::ThreadStateSliceCallstack>,
-    std::pair<SchedSwitchWithStackPerfEvent, orbit_grpc_protos::ThreadStateSliceCallstack>>;
-INSTANTIATE_TYPED_TEST_SUITE_P(UprobesUnwindingVisitorSampleTestNew,
+    DwarfUnwindingTestType<StackSamplePerfEvent, orbit_grpc_protos::FullCallstackSample>,
+    DwarfUnwindingTestType<SchedWakeupWithStackPerfEvent,
+                           orbit_grpc_protos::ThreadStateSliceCallstack>,
+    DwarfUnwindingTestType<SchedSwitchWithStackPerfEvent,
+                           orbit_grpc_protos::ThreadStateSliceCallstack>>;
+INSTANTIATE_TYPED_TEST_SUITE_P(UprobesUnwindingVisitorDwarfUnwindingTest,
```

Comment:
Ah another thing. The point of the prefix is that you could declare `INSTANTIATE_TYPED_TEST_SUITE_P` multiple times with different sets of types, and the prefix allows to distinguish by "describing" the collection of types. You can see this in sample6_unittest.cc linked in [go/gunitadvanced](https://goto.google.com/gunitadvanced). So I don't think we should repeat the name of the fixture. 

But since we only declare this once, we should probably just use something simple. [go/gunitadvanced](https://goto.google.com/gunitadvanced) itself simply uses "My". I would also just leave it empty but I don't know if that's possible.

Reply:
Went for "TypedTest"